src/test_l1_dsFPD.c

```diff
@@ -107,25 +103,29 @@ static bool extendedEnumsSupported=false; //Default to not supported
     }\
 }\
 
-
-#define enableFPDIndicators(){\
+#define enableFPDIndicators(count) {\
     dsError_t result;\
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)\
-    {\
-        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, kIndicators[i].id);\
-        result = dsSetFPState(kIndicators[i].id, dsFPD_STATE_ON);\
+    dsFPDIndicator_t eIndicator;\
+    char buffer[DS_FPD_KEY_SIZE];\
+    for (int i = 0; i < (count); i++) {\
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);\
```

Comment:
According to test profile files, the fp indicators start from 1, but according to the present scenario, it starts from 0, which is unavailable.  There are multiple places with similar implementation which needs to be addressed.

Reply:
Done

Reply:
sumi please check the i value since we are getting count value also 1 in that case  for loop will not execute.

Reply:
Done